chain/network/src/routing/graph_with_cache.rs

```diff
@@ -110,9 +133,24 @@ impl GraphWithCache {
             })
             .cloned()
             .collect();
-        for edge in &edges {
-            self.remove_edge(edge.key());
-        }
+        self.remove_edges(&edges.iter().map(|edge| edge.key()).collect::<Vec<_>>());
         edges
     }
+
+    pub fn prune_old_edges(&mut self, prune_edges_older_than: time::Utc) {
+        self.prune_edges_before = Some(prune_edges_older_than);
+        let old_edges = self
+            .edges()
+            .iter()
+            .filter_map(|(edge_key, edge)| {
+                if edge.is_edge_older_than(prune_edges_older_than) {
+                    Some(edge_key)
+                } else {
+                    None
+                }
+            })
+            .cloned()
+            .collect::<Vec<_>>();
+        self.remove_edges(&old_edges.iter().map(|key| key).collect::<Vec<_>>());
```

Comment:
```suggestion
        self.remove_edges(&old_edges.iter().collect::<Vec<_>>());
```

or is there some coercion here?